bson/src/test/unit/org/bson/codecs/IterableCodecSpecification.groovy

```diff
@@ -176,7 +154,7 @@ class IterableCodecSpecification extends Specification {
         where:
         representation | value                                                     | document
         STANDARD       | [UUID.fromString('01020304-0506-0708-090a-0b0c0d0e0f10')] | '{"array": [{ "$binary" : "AQIDBAUGBwgJCgsMDQ4PEA==", "$type" : "4" }]}'
-        JAVA_LEGACY    | [UUID.fromString('01020304-0506-0708-090a-0b0c0d0e0f10')] | '{"array": [{ "$binary" : "AQIDBAUGBwgJCgsMDQ4PEA==", "$type" : "4" }]}'
+        JAVA_LEGACY    | [UUID.fromString('01020304-0506-0708-090a-0b0c0d0e0f10')] | '{"array": [{ "$binary" : "CAcGBQQDAgEQDw4NDAsKCQ==", "$type" : "3" }]}'
```

Comment:
So JAVA_LEGACY is now subtype 3?

Reply:
All the legacy ones are subtype 3.  This change in the test corresponds to the change in the condition for type 4 binaries in IterableCode.readValue,

Reply:
Ack